extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmProcessor.java

```diff
@@ -189,21 +189,25 @@ public final class HibernateOrmProcessor {
     void registerHibernateOrmMetadataForCoreDialects(
             BuildProducer<DatabaseKindDialectBuildItem> producer) {
         producer.produce(new DatabaseKindDialectBuildItem(DatabaseKind.DB2,
-                "org.hibernate.dialect.DB297Dialect"));
+                "org.hibernate.dialect.DB2Dialect"));
         producer.produce(new DatabaseKindDialectBuildItem(DatabaseKind.DERBY,
-                "org.hibernate.dialect.DerbyTenSevenDialect"));
+                "org.hibernate.dialect.DerbyDialect"));
         producer.produce(new DatabaseKindDialectBuildItem(DatabaseKind.H2,
                 "io.quarkus.hibernate.orm.runtime.dialect.QuarkusH2Dialect"));
+        // using MariaDBDialect will make us go back to 10.3
         producer.produce(new DatabaseKindDialectBuildItem(DatabaseKind.MARIADB,
                 "org.hibernate.dialect.MariaDB106Dialect"));
+        // using SQLServerDialect will make us go back to 2008
         producer.produce(new DatabaseKindDialectBuildItem(DatabaseKind.MSSQL,
                 "org.hibernate.dialect.SQLServer2016Dialect"));
+        // using MySQLDialect will make us go back to 5.7
         producer.produce(new DatabaseKindDialectBuildItem(DatabaseKind.MYSQL,
                 "org.hibernate.dialect.MySQL8Dialect"));
+        // using OracleDialect will make us go back to 11.2
         producer.produce(new DatabaseKindDialectBuildItem(DatabaseKind.ORACLE,
                 "org.hibernate.dialect.Oracle12cDialect"));
         producer.produce(new DatabaseKindDialectBuildItem(DatabaseKind.POSTGRESQL,
-                "io.quarkus.hibernate.orm.runtime.dialect.QuarkusPostgreSQL10Dialect"));
+                "org.hibernate.dialect.PostgreSQLDialect"));
```

Comment:
The new ORM 6 dialect includes the specific handling we had in the Quarkus one so we can just drop it.
It was in the `runtime` package so not supposed to be used publicly so I think we can drop them.

Reply:
Need to be careful here: remember Quarkus doesn't get to connect to the DB to match the version of the running Database.
So this change is essentially setting the PostgreSQL dialect to run with the minimal version supported by ORM, which is not the same as the minimal version expected by Quarkus. This will change the syntax and the supported capabilities.

Reply:
See the other comments. I checked for all of them that the minimum version was at least the one we were using in Quarkus. Thus why I haven't changed all of them and added comments.